### ISSUE TITLE
Drush no longers works without write access to the users homefolder

### ISSUE THREAD

**Issue by sandervd** [drush-ops/drush#3221]
The refactoring in 9.0.0-rc (issue is not present in 9.0.0-beta9) causes drush to fail with the following error:
>  [error]  Directory /home/ubuntu exists, but is not writable. Please check directory permissions.

This is especially problematic for automated deployment, where the deployment might run as a system user that lacks a home directory.

**Comment by greg-1-anderson** [comment]
This would be good to fix. There weren't too many changes in rc1

**Comment by weitzman** [comment]
Not sure if we should fix this as there are numerous workaround. The first
is that you can the a HOME env variable to somewhere writable. Here is the
code from \Webmozart\PathUtil\Path::getHomeDirectory.

```
// For UNIX support
        if (getenv('HOME')) {
            return static::canonicalize(getenv('HOME'));
        }

        // For >= Windows8 support
        if (getenv('HOMEDRIVE') && getenv('HOMEPATH')) {
            return
static::canonicalize(getenv('HOMEDRIVE').getenv('HOMEPATH'));
        }
```

**Comment by weitzman** [comment]
What should we do if there is no HOME? Use tmp? What if there is no writable tmp?

**Comment by claudiu-cristea** [comment]
> What should we do if there is no HOME? Use tmp? What if there is no writable tmp?

Not having `/tmp` (in fact `sys_get_temp_dir()`) it seems to be very unlikely. So, for such cases let's popup a warning and disable cache (not sure Drush works w/o cache).

**Comment by claudiu-cristea** [comment]
Fixed by #3232.
